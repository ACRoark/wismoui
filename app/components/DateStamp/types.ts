export type DateStampFormat = 'auto' | 'long' | 'short';
