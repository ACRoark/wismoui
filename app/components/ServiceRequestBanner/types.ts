export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
