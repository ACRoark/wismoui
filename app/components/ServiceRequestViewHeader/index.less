@import (reference) '../../styles/themes/index.less';

@service-request-view-header-cls-prefix: ~'@{dte-prefix}-service-request-view-header';

.@{service-request-view-header-cls-prefix} {
  border-bottom: 1px solid @dte-gray-300;
  padding-bottom: 25px;
  padding-top: 25px;
  width: 100%;

  &-content {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    @media @mobile {
      align-items: flex-start;
      flex-direction: column;
    }
  }

  .back-link {
    font-weight: @dte-wismo-font-weight-semi-bold;

    > a {
      align-items: center;
      display: flex;

      &:link {
        color: @dte-blue-light;
        text-decoration: none;
      }

      &:visited {
        color: @dte-blue-light;
      }

      &:hover {
        color: @dte-interactive-hover;
        text-decoration: none !important;
      }

      .back-arrow {
        margin-right: 11px;
      }
    }

    @media @mobile {
      padding-bottom: 17px;
    }
  }

  .service-request-number {
    color: @dte-gray-700;
    text-align: right;

    .label {
      color: @dte-gray-700;
      font-size: @dte-wismo-font-size-1;
      font-weight: @dte-wismo-font-weight-semi-bold;
      padding-right: 6px;
    }
  }
}
