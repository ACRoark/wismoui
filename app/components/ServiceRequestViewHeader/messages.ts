import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.service-request-view-header';

export default defineMessages({
  orderNumber: {
    id: `${scope}.order-number`,
  },
  backToAllOrders: {
    id: `${scope}.back-to-all-orders`,
  },
  backToSearchPage: {
    id: `${scope}.back-to-search-page`,
  },
});
