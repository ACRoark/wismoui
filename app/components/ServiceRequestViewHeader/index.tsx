import React from 'react';
import { FormattedMessage } from 'react-intl';

import ColumnLayout from 'components/ColumnLayout';
import Hyperlink from 'components/Hyperlink';
import OrderNumber from 'components/OrderNumber';
import useSearch from 'hooks/useSearch';
import BackArrow from 'images/BackArrow';
import { userIsAuthenticated } from 'security/utils';

import messages from './messages';

import './index.less';

interface IServiceRequestViewHeaderProps {
  serviceRequestNumber: string;
}

const ServiceRequestViewHeader: React.FC<IServiceRequestViewHeaderProps> = (
  props: IServiceRequestViewHeaderProps,
): React.ReactElement => {
  const { serviceRequestNumber } = props;
  const { results = [] } = useSearch();
  const userHasMultipleOrders = results.length > 1;

  const renderBackLink = (isAuthenticated: boolean, multipleOrders: boolean): React.ReactElement => {
    if (isAuthenticated) {
      return multipleOrders ? (
        <div className="back-link" data-testid="back-link">
          <Hyperlink href="/">
            <BackArrow />
            <FormattedMessage {...messages.backToAllOrders} />
          </Hyperlink>
        </div>
      ) : (
        <div />
      );
    }

    return (
      <div className="back-link" data-testid="back-link">
        <Hyperlink href="/">
          <BackArrow />
          <FormattedMessage {...messages.backToSearchPage} />
        </Hyperlink>
      </div>
    );
  };

  return (
    <div className="dte-wismo-service-request-view-header">
      <ColumnLayout>
        <div className="dte-wismo-service-request-view-header-content">
          {renderBackLink(userIsAuthenticated(), userHasMultipleOrders)}
          <div className="service-request-number">
            <span className="label">
              <FormattedMessage {...messages.orderNumber} />
            </span>
            <OrderNumber value={serviceRequestNumber} />
          </div>
        </div>
      </ColumnLayout>
    </div>
  );
};

export default ServiceRequestViewHeader;
