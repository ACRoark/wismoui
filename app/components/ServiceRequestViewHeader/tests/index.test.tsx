import { RouterState } from 'connected-react-router';
import { mock } from 'jest-mock-extended';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import { SUPPORTED_LOCALES } from 'locales';

import { userIsAuthenticated } from 'security/utils';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { IApplicationConfig, IApplicationRootState, IAuthenticationState, ISearchState, IStartupState } from 'types';

import ServiceRequestViewHeader from '..';

jest.mock('security/utils');

const mockAuth = userIsAuthenticated as jest.MockedFunction<typeof userIsAuthenticated>;

const orderNumbers = ['MI12345678', 'MO12345678', 'MT12345678'];

const multipleSearchResultsState: IApplicationRootState = {
  auth: mock<IAuthenticationState>(),
  config: mock<IApplicationConfig>(),
  router: mock<RouterState>(),
  search: mock<ISearchState>({
    error: undefined,
    loading: false,
    results: [
      {},
      {}
      ],
  }),
  startup: mock<IStartupState>(),
  tracker: {
    error: undefined,
    loading: false,
    order: undefined,
  },
};

const singleSearchResultState: IApplicationRootState = {
  auth: mock<IAuthenticationState>(),
  config: mock<IApplicationConfig>(),
  router: mock<RouterState>(),
  search: mock<ISearchState>({
    error: undefined,
    loading: false,
    results: [{}],
  }),
  startup: mock<IStartupState>(),
  tracker: {
    error: undefined,
    loading: false,
    order: undefined,
  },
};

const expectComponentToMatchSnapshot = (isAuthenticated: boolean, language: string, orderNumber: string, searchResults: IApplicationRootState): void => {
  mockAuth.mockReturnValue(isAuthenticated);
  const store = createMockStore(searchResults);

  const tree = createSnapshotWithIntl(
    <Router>
      <Provider store={store}>
        <ServiceRequestViewHeader serviceRequestNumber={orderNumber}/>
      </Provider>
    </Router>, language);

  expect(tree).toMatchSnapshot();
};

const shouldNotLogErrors = (isAuthenticated: boolean, language: string, orderNumber: string, searchResults: IApplicationRootState): void => {
  mockAuth.mockReturnValue(isAuthenticated);
  const store = createMockStore(searchResults);

  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(
    <Router>
      <Provider store={store}>
        <ServiceRequestViewHeader serviceRequestNumber={orderNumber}/>
      </Provider>
    </Router>, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('<ServiceRequestViewHeader />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    orderNumbers.forEach((orderNumber: string): void => {
      it(`should not log errors when rendered in ${language} for order ${orderNumber}, search has many results and user is authenticated`, (): void => {
        shouldNotLogErrors(true, language, orderNumber, multipleSearchResultsState);
      });

      it(`should not log errors when rendered in ${language} for order ${orderNumber}, search has one result and user is authenticated`, (): void => {
        shouldNotLogErrors(true, language, orderNumber, singleSearchResultState);
      });

      it(`should not log errors when rendered in ${language} for order ${orderNumber}, search has many results and user is not authenticated`, (): void => {
        shouldNotLogErrors(false, language, orderNumber, multipleSearchResultsState);
      });

      it(`should not log errors when rendered in ${language} for order ${orderNumber}, search has one result and user is not authenticated`, (): void => {
        shouldNotLogErrors(false, language, orderNumber, singleSearchResultState);
      });

      // tslint:disable-next-line:max-line-length
      it(`should render correctly for order number ${orderNumber} when the current language is ${language}, search has many results, and user is authenticated`, (): void => {
        expectComponentToMatchSnapshot(true, language, orderNumber, multipleSearchResultsState);
      });

      // tslint:disable-next-line:max-line-length
      it(`should render correctly for order number ${orderNumber} when the current language is ${language}, search has one results, and user is authenticated`, (): void => {
        expectComponentToMatchSnapshot(true, language, orderNumber, singleSearchResultState);
      });

      it(`should render correctly for order number ${orderNumber} when the current language is ${language}, search has many results, and user is not authenticated`, (): void => {
        expectComponentToMatchSnapshot(false, language, orderNumber, multipleSearchResultsState);
      });

      it(`should render correctly for order number ${orderNumber} when the current language is ${language}, search has one result, and user is not authenticated`, (): void => {
        expectComponentToMatchSnapshot(false, language, orderNumber, singleSearchResultState);
      });
    });
  });
});
