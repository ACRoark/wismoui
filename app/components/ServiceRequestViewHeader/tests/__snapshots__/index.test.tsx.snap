// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is en, search has many results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Back to All Orders
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is en, search has many results, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Search Another Order
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is en, search has one result, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Search Another Order
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is en, search has one results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div />
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is es, search has many results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Volver a todos los pedidos
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is es, search has many results, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Buscar otra orden
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is es, search has one result, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Buscar otra orden
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MI12345678 when the current language is es, search has one results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div />
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MI 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is en, search has many results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Back to All Orders
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is en, search has many results, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Search Another Order
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is en, search has one result, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Search Another Order
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is en, search has one results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div />
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is es, search has many results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Volver a todos los pedidos
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is es, search has many results, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Buscar otra orden
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is es, search has one result, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Buscar otra orden
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MO12345678 when the current language is es, search has one results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div />
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MO 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is en, search has many results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Back to All Orders
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is en, search has many results, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Search Another Order
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is en, search has one result, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Search Another Order
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is en, search has one results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div />
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is es, search has many results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Volver a todos los pedidos
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is es, search has many results, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Buscar otra orden
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is es, search has one result, and user is not authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div
        className="back-link"
        data-testid="back-link"
      >
        <a
          href="/"
          onClick={[Function]}
          target="_self"
        >
          <svg
            className="back-arrow"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.354 1.646a.5.5 0 010 .708L5.707 8l5.647 5.646a.5.5 0 01-.708.708l-6-6a.5.5 0 010-.708l6-6a.5.5 0 01.708 0z"
              fillRule="evenodd"
              stroke="currentColor"
              strokeWidth="1"
            />
          </svg>
          Buscar otra orden
        </a>
      </div>
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ServiceRequestViewHeader /> should render correctly for order number MT12345678 when the current language is es, search has one results, and user is authenticated 1`] = `
<div
  className="dte-wismo-service-request-view-header"
>
  <div
    className="dte-wismo-column-layout"
  >
    <div
      className="dte-wismo-service-request-view-header-content"
    >
      <div />
      <div
        className="service-request-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span>
          MT 1234 5678
        </span>
      </div>
    </div>
  </div>
</div>
`;
